Cargo.toml

```diff
@@ -10,6 +18,179 @@ members = [
 	"runtime/*",
 ]
 
+[workspace.dependencies]
+# crates.io
+array-bytes       = { version = "6.0" }
```

Comment:
As I saw, upstream didn't replace this field.

How to check if a dependency is never used?

Reply:
> As I saw, upstream didn't replace this field.

The frontier did. This is mainly a rust usage refactor, no logic was touched. This makes crate upgrade much easier.

> How to check if a dependency is never being used?

This PR's main aim is to reduce the duplication of the crate's dependencies and metadata. No improvement in detecting whether a dependency is being used.

Reply:
I mean, once we remove a dependency on a single crate, how to make sure that dependency isn't being used in other crates?

Reply:
> how to make sure that dependency isn't being used in other crates?

Global search

> With these changes, we can not easily find that out.

Without a global search, how did you find this before?

Reply:
I don't need to global search. Just remove the unused dependency from the crate's toml file.

Reply:
If `darwinia-staking` don't need `sp-io` anymore. Then I can remove it from the `darwinia-staking/Cargo.toml`. Now, I need to search if other pallets are using `sp-io`, then I can remove it from the `root/Cargo.toml`.